core/ppu.cpp

```diff
@@ -212,15 +214,21 @@ void PPU::HandleCpuWrite( u16 address, u8 data ) // NOLINT
                    and is set to bits 8-14 of _tempAddr
                    The _addrLatch is toggled
                  */
-                // TODO: Implement
+                _tempAddr.bit.coarseX = data & 0x1F; // shifts for high coarseX, nametablex/y
+                _tempAddr.bit.nametableX = ( data & 0x40 ) >> 6;
+                _tempAddr.bit.nametableY = ( data & 0x80 ) >> 7;
+                _addrLatch = true;
```

Comment:
I don't think this works as expected. The first write is supposed to set bits 8-14, without modifying bits 0-7. coarseX is the lower 5 bits, while nametable x and nametable y are bits 10 and 11. You should instead update _tempAddr.value directly. 

1. AND _tempAddr.value with 0xFF to preserve bits 0-7.
2. AND together data and 0x7F to grab bits 0-6 from data.
3. shift the result of the previous operation left 8 bits
4. OR together the result of the previous operation and the result of step 1
5. Assign the new result to _tempAddr.value

```suggestion
                _tempAddr.value = ( _tempAddr.value & 0xFF ) | (( data & 0x7F ) << 8 ) 
                _addrLatch = true;
```